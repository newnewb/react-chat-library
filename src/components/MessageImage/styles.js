// @ts-check

const styles = {
	container: {},
	image: {
		width: 150,
		height: 100,
		borderRadius: 13,
		margin: 3,
		resizeMode: 'cover'
	},
	imageActive: {
		flex: 1,
		resizeMode: 'contain'
	}
};

export default styles;
