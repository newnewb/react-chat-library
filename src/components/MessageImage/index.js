// @ts-check

// Packages
import React from 'react';

// Ours
import { CSS } from '../../utils';
import styles from './styles';

// TODO: support lightbox like functionality

/**
 * @typedef Props
 * @property {string} source
 * @property {string} [alt]
 * @property {object} [containerStyle]
 * @param {Props} props
 */
const MessageImage = props => (
	<div style={CSS([styles.container, props.containerStyle])}>
		<img src={props.source} alt={props.alt} />
	</div>
);

/** @type Props */
MessageImage.defaultProps = {
	source: null,
	alt: '',
	containerStyle: {}
};

export default MessageImage;
