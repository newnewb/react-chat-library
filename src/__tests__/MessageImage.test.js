import React from 'react';
import renderer from 'react-test-renderer';

import MessageImage from '../components/MessageImage';

it('should render <MessageImage /> and compare with snapshot', () => {
	const component = renderer.create(<MessageImage />);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
