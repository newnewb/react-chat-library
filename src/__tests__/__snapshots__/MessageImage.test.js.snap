// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <MessageImage /> and compare with snapshot 1`] = `
<div
  style={Object {}}
>
  <img
    alt=""
    src={null}
  />
</div>
`;
